app/assets/src/components/PipelineSampleTree.jsx

```diff
@@ -78,13 +81,15 @@ class PipelineSampleTree extends React.PureComponent {
       for (let j = 0; j < order.length; j += 1) {
         let level = order[j],
           taxon_id = row.lineage[level + "_taxid"],
+          is_common_name = false,
           name;
 
-        if (this.props.nameType == "Common name") {
-          name = row.lineage[level + "_common_name"];
-        }
-
-        if (!name) {
+        if (
+          this.props.nameType == NAME_TYPE_COMMON_NAME &&
+          (name = row.lineage[level + "_common_name"])
```

Comment:
wait did you mean 'name == row.lineage'? or should that line be in the 'if' block?